samples/tutorials/mnist/04_Reusable_and_Pre-build_Components_as_Pipeline.ipynb

```diff
@@ -0,0 +1,773 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Copyright 2019 Google Inc. All Rights Reserved.\n",
+    "#\n",
+    "# Licensed under the Apache License, Version 2.0 (the \"License\");\n",
+    "# you may not use this file except in compliance with the License.\n",
+    "# You may obtain a copy of the License at\n",
+    "#\n",
+    "#      http://www.apache.org/licenses/LICENSE-2.0\n",
+    "#\n",
+    "# Unless required by applicable law or agreed to in writing, software\n",
+    "# distributed under the License is distributed on an \"AS IS\" BASIS,\n",
+    "# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n",
+    "# See the License for the specific language governing permissions and\n",
+    "# limitations under the License.\n",
+    "# =============================================================================="
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Reusable components and Pre-build components\n",
+    "\n",
+    "This tutorial describes the manual way of writing a full component program (in any language) and a component definition for it. Below is a summary of the steps involved in creating and using a component:\n",
+    "\n",
+    "- Write the program that contains your component’s logic. The program must use files and command-line arguments to pass data to and from the component.\n",
+    "- Containerize the program.\n",
+    "- Write a component specification in YAML format that describes the component for the Kubeflow Pipelines system.\n",
+    "- Use the Kubeflow Pipelines SDK to load your component, use it in a pipeline and run that pipeline.\n",
+    "\n",
+    "More over, we will combine our built components together with a pre-build components and a lightweight component to compose a pipeline with three steps\n",
+    "- Train a minist model and export to GCS\n",
+    "- Deploy the exported tensorflow model on AI Platform\n",
+    "- Test the deployment by calling the end point\n",
+    "\n",
+    "**Note: Make sure that you have docker installed in the local environment if you want to build the image locally**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
```

Comment:
You might want to consider using `%load` magic to reduce duplicated code across noteboooks

Reply:
I would prefer to keep each notebook as standalone as much possible. It's also easier when people just look at the examples while browsing in github directly. Hope this makes sense :)